### PR TITLE
fix: [CDS-69449]: handle the case when items in repeat looping strategy is an expression in CI stage

### PR DESCRIPTION
**Pull request by shalini-agr**
## Describe your changes

**Issue:** When an expression is used for items in a repeat looping strategy in a CI step, its value is resolved as textNode in ExecutionWrapperConfig step node and so, during its deserialization, we get a mismatch exception that string cannot be converted to List<String>.

**Changes:** Handled the case in ParameterFieldDeserializer when the value is a textNode with the stringified list and needs to convert into List<String>

**Testing:** Tested this locally: Have a CI step with repeat looping strategy and its items as an expression with .split(","). The pipeline should be executed successfully with a looping strategy. Also, tested the basic case when the items are a fixed value and that error should be thrown if the items are fixed string instead of list.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49507)

### CODE REVIEW

File: 953-yaml-commons/src/main/java/io/harness/serializer/jackson/CustomStringListDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.List;
+
+@OwnedBy(HarnessTeam.PIPELINE)
+public class CustomStringListDeserializer extends StdDeserializer<ParameterField<List<String>>> {
+  public CustomStringListDeserializer() {
+    super(ParameterField.class);
+  }
+
+  @Override
+  public ParameterField<List<String>> deserialize(JsonParser jsonParser, DeserializationContext deserializationContext)
```
Comment (mohitg0795) [inline 953-yaml-commons/src/main/java/io/harness/serializer/jackson/CustomStringListDeserializer.java:36]: a UT would be great to have here

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 8 of 8 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49507)* status: 9 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`953-yaml-commons/src/test/java/io/harness/yaml/utils/JsonPipelineUtilsTest.java` line 36 at r3](https://reviewable.io/reviews/harness/harness-core/49507#-NYbQ47Q84YWK71mV3mM:-NYbQ47Q84YWK71mV3mN:bjsoi6y) ([raw file](https://github.com/harness/harness-core/blob/9a5165838b3bb63c46e3e92d6c93286cf179ded9/953-yaml-commons/src/test/java/io/harness/yaml/utils/JsonPipelineUtilsTest.java#L36)):*
> ```Java
>         + "  items: \"[\\\"a\\\",\\\"b\\\"]\"";
>     StrategyConfig strategyConfig =
>         JsonPipelineUtils.read(YamlUtils.readAsJsonNode(yaml).toString(), StrategyConfig.class);
> ```

what is this test checking?

**Review by shalini-agr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49507)* status: 9 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`953-yaml-commons/src/test/java/io/harness/yaml/utils/JsonPipelineUtilsTest.java` line 36 at r3](https://reviewable.io/reviews/harness/harness-core/49507#-NYbQ47Q84YWK71mV3mM:-NYbQK6c9FNdS0ZYD4Mh:b-pw31gv) ([raw file](https://github.com/harness/harness-core/blob/9a5165838b3bb63c46e3e92d6c93286cf179ded9/953-yaml-commons/src/test/java/io/harness/yaml/utils/JsonPipelineUtilsTest.java#L36)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

what is this test checking?
</blockquote></details>

in the repeat, we have given items as a string with its value as a list. Without the change, this test will fail with mismatchedInputException similar to the use case described in the ticket.

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49507)* status: 9 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by brijeshdhakar — DISMISSED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, 8 of 8 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49507)* status: 1. unresolved discussions.

**Comment by shalini-agr** [comment]
trigger uts

**Review by archit-harness — APPROVED** [review]
> Reviewed 6 of 6 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49507)* status: 6 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by brijeshdhakar — APPROVED** [review]
> Reviewed 6 of 6 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49507)* status: 1. unresolved discussions.

**Pull request merged**